src/components/CreateUserForm/index.tsx

```diff
@@ -0,0 +1,138 @@
+import { useForm, SubmitHandler } from 'react-hook-form';
+
+type FormValues = {
+  firstName: string;
+  lastName: string;
+  role: string;
+  phoneNumber: number;
+  email: string;
+};
+
+const CreateUserForm = () => {
+  const { register, handleSubmit } = useForm<FormValues>();
+
+  const onSubmit: SubmitHandler<FormValues> = async (data) => {
+    const BASE_URL = process.env.ITEZ_API_URI;
+
+    try {
+      const res = await fetch(`${BASE_URL}/user`, {
+        body: JSON.stringify(data),
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        method: 'POST',
+      });
```

Comment:
perhaps these headers will be repeatitive in so many places,
we might as well have it as  a reusable CONSTANT